src/main/java/lottery/InfoCenter.java

```diff
@@ -1,23 +1,29 @@
 package lottery;
 
+import static java.util.Objects.*;
 import static java.util.stream.Collectors.*;
 
-import java.util.stream.Collectors;
-
 public class InfoCenter {
 
-	private final int MATCH_THREE_NUMBER = 3;
 	private Ticket lastWeekWinningTicket;
+	private int bonus;
 
 	public void setLastWeekWinningTicket(Ticket lastWeekWinningTicket) {
 		this.lastWeekWinningTicket = lastWeekWinningTicket;
 	}
 
+	void setBonusNumber(int number) {
+		this.bonus = number;
+	}
+
 	public Result confirmTicket(Tickets buyerTickets) {
+		requireNonNull(lastWeekWinningTicket, "지난 주 티켓이 없습니다.");
+		requireNonNull(bonus, "보너스 볼이 없습니다.");
+
 		return new Result(LotteryMatchTypeMap.of(buyerTickets.getValues().stream()
-			.map(ticket -> ticket.numbers().matchCountWith(lastWeekWinningTicket.numbers()))
-			.filter(matchCount -> matchCount >= MATCH_THREE_NUMBER)
-			.collect(Collectors.groupingBy(LotteryMatchType::fromInteger, summingInt(a -> 1)))));
+			.map(ticket -> ticket.numbers().getMatchTypeWith(lastWeekWinningTicket.numbers(), bonus))
```

Comment:
규칙 4: 한 줄에 점을 하나만 찍는다.

위 객체지향 생활 체조 원칙 관점에서 봤을 때 아래와 같이 호출되는건 어떨까요?

```suggestion
			.map(ticket -> ticket.getMatchTypeWith(lastWeekWinningTicket.numbers(), bonus))
```